api/api-pipeline-operations-v1/src/main/java/com/thoughtworks/go/apiv1/pipelineoperations/representers/EnvrionmentVariableRepresenter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.pipelineoperations.representers;
+
+import com.google.gson.JsonParseException;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.api.representers.JsonWriter;
+import com.thoughtworks.go.server.service.PipelineTriggerService;
+import com.thoughtworks.go.spark.RequestContext;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.Map;
+
+public class EnvrionmentVariableRepresenter {
+    public static Map toJSON(PipelineTriggerService.EnvironmentVariable environmentVariable, RequestContext requestContext) {
+        if (environmentVariable == null) return null;
+        JsonWriter jsonWriter = new JsonWriter(requestContext);
+        jsonWriter.add("name", environmentVariable.name());
+        jsonWriter.add("value", environmentVariable.value());
+        jsonWriter.add("encrypted_value", environmentVariable.encryptedValue());
+        jsonWriter.add("secure", environmentVariable.isSecure());
+        return jsonWriter.getAsMap();
+    }
+
+    public static PipelineTriggerService.EnvironmentVariable fromJSON(JsonReader jsonReader) {
+        if (jsonReader == null) {
+            return null;
+        }
+        String name = jsonReader.getString("name");
+        String value = jsonReader.optString("value").orElse(null);
+        Boolean secure = jsonReader.optBoolean("secure").orElse(false);
+        String encryptedValue = jsonReader.optString("encrypted_value").orElse(null);
+        if (!StringUtils.isBlank(value) && !StringUtils.isBlank(encryptedValue))
+            throw new JsonParseException(String.format("Environment variable '%s' has both value and encryptedValue defined.", name));
+        return new PipelineTriggerService.EnvironmentVariable(name, value, encryptedValue, secure);
```

Comment:
Was there a reason to not use `EnvironmentVariableConfig` ? it already has most of these things and can validate itself during `EnvironmentVariableConfig.deserialize` with lot more edge cases.